src/Microsoft.Framework.ApplicationHost/Impl/Syntax/CommandGrammar.cs

```diff
@@ -30,6 +30,12 @@ public CommandGrammar(Func<string, string> variable)
             var unquotedTerm = Rep1(unquotedPiece.Or(specialPiece)).Str();
 
             var quotedTerm = Ch('\"').And(Rep(quotedPiece.Or(specialPiece)).Str()).And(Ch('\"')).Left().Down();
+            if (preserveSurroundingQuotes)
+            {
+                // Str() value assigned to quotedTerm does not include quotation marks surrounding the quoted or
+                // special piece. Add those quotes back if requested.
+                quotedTerm = quotedTerm.Build(str => "\"" + str + "\"");
+            }
```

Comment:
Also likely minor but `quotedTerm` will not keep a (Linux-only) single-quoted argument string together.